pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -191,7 +192,9 @@ type AWSCloudConfig struct {
 	Global struct {
 		// TODO: Is there any use for this?  We can get it from the instance metadata service
 		// Maybe if we're not running on AWS, e.g. bootstrap; for now it is not very useful
-		Zone string
+		Zone     string
+		Region   string
+		Endpoint string
```

Comment:
Is there any way to get the endpoint from the metadata service on your AWS implementation?  And which one are you using (we might actually be able to test against it, which could be cheaper than testing against AWS!)